### PR TITLE
fix bug: for files that don't change but added/removed from docfx.json, changes should be tracked

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs [hunk 1]
```diff
@@ -180,18 +180,29 @@ public void LoadChanges()
...
+            var lastFileAttributes = LastBuildVersionInfo.Attributes;
             if (_parameters.Changes != null)
             {
                 // use user-provided changelist
                 foreach (var pair in _parameters.Changes)
                 {
                     _changeDict[pair.Key] = pair.Value;
                 }
+
+                //// scenario: file itself doesn't change but add/remove from docfx.json
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs]: >//// [](start = 16, length = 4)

minor: //

**Comment by hellosnow** [comment]
:shipit:

File: test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs [hunk 2]
```diff
@@ -607,6 +607,199 @@ public void TestServerChanges()
             }
         }
 
+        [Fact]
+        public void TestFilesAddRemoveFromDocfx()
+        {
+            // conceptual1--->conceptual2(phase 2)
+            // conceptual2--->conceptual3(phase 1)
+            // conceptual3
+            // conceptual4
```
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: update

File: test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs [hunk 3]
```diff
@@ -607,6 +607,199 @@ public void TestServerChanges()
...
+                            ["meta"] = "Hello world!",
+                        },
+                        templateFolder: templateFolder,
+                        intermediateFolder: intermediateFolder,
+                        changes: new Dictionary<string, ChangeKindWithDependency>
+                        {
+                            { ((RelativePath)conceptualFile3).GetPathFromWorkingFolder(), ChangeKindWithDependency.Updated },
+                            { ((RelativePath)conceptualFile).GetPathFromWorkingFolder(), ChangeKindWithDependency.None },
+                            { ((RelativePath)conceptualFile4).GetPathFromWorkingFolder(), ChangeKindWithDependency.None },
+                            { ((RelativePath)conceptualFile5).GetPathFromWorkingFolder(), ChangeKindWithDependency.None },
```
Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: >None [](start = 115, length = 4)

Created?
And file2 is Deleted?

Comment (ansyral) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: for server, they only track whether the file changes.
the case is to test the scenario that file doesn't change, but is added/removed from docfx. so the changelist doesn't contain the files.

Comment (superyyrrzz) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: Got it. So should file12345 all be ChangeKindWithDependency.None?

Comment (ansyral) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/IncrementalBuildTest.cs]: i also tested with 3 updated

---
In reply to: [93399298](https://github.com/dotnet/docfx/pull/1031#discussion_r93399298) [](ancestors = 93399298)

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**